File: problems/problems_2779/Solution.java
package problems.problems_2779;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int maximumBeauty(int[] nums, int k) {
        int m = 0;
        for (int num: nums) {
            m = Math.max(m, num);
        }
        int[] diffs = new int[m + 2];
        for (int num: nums) {
            diffs[Math.max(0, num - k)]++;
            diffs[Math.min(m + 1, num + k + 1)]--;
        }
        int ans = 0, cur = 0;
        for (int v: diffs) {
            cur += v;
            ans = Math.max(ans, cur);
        }
        return ans;
    }

    @Override
    public Object solve(String[] values) {
        int[] nums = jsonArrayToIntArray(values[0]);
		int k = Integer.parseInt(values[1]);
        return JSON.toJSON(maximumBeauty(nums, k));
    }
}

File: problems/problems_2779/solution.py
import solution
from typing import *
from collections import defaultdict


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maximumBeauty(*test_input)

    def maximumBeauty(self, nums: List[int], k: int) -> int:
        diffs = defaultdict(int)
        for num in nums:
            diffs[num - k] += 1
            diffs[num + k + 1] -= 1
        ans = cur = 0
        for _, v in sorted(diffs.items()):
            cur += v
            ans = max(ans, cur)
        return ans
